AirPal/RootView.swift

```diff
@@ -26,12 +27,30 @@ struct RootView: View {
                 Text(viewModel.flightData?.arrival.iata ?? "Ari")
             }
             Text("Flight Status: \(viewModel.flightData?.flightStatus ?? "N/A")")
+            Button("Refresh") {
+                Task {
+                    do {
+                        flightChanged = try await viewModel.refreshFlightData()
+                    } catch {
+                        print("Error refreshing", error)
+                    }
+                }
+            }
+            if flightChanged {
+                VStack(alignment: .center) {
+                    ForEach(viewModel.messages, id: \.self) {
+                        Text($0)
+                            .multilineTextAlignment(.center)
+                    }
+                }
+                .padding(.horizontal)
+            }
         }
         .padding()
         .onSubmit(of: .text) {
             Task {
                 do {
-                    try await viewModel.getFlightData(flightNumber: flightInput.uppercased())
+                    viewModel.flightData =  try await viewModel.getFlightData(flightNumber: flightInput.uppercased())
```

Comment:
May I suggest a teensy pivot on the flow here? 

Current flow:
1. View has flightNumber
2. View asks ViewModel to get data with flightNumber
3. ViewModel gives data back
4. View tells ViewModel the flight data
5. View reads flihgt data from ViewModel elsewhere in the view

Who should own flightData? What signifies data should be in the View vs. the ViewModel? 

It's a small thing, but I think it can help make a muuuch clearer mental model of how the View works :)

Reply:
Feel free to iterate on the next PR if this is a lot to fit in one PR too btw!

Reply:
resolved in pairing! :)